### PR TITLE
created settings screen with async storage syncronization

### PR DESCRIPTION
**Pull request by MusheghKh**

### CODE REVIEW

File: app/SettingsScreen.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+	componentDidMount(){
+		this.storageToState();
+	}
+
+	storageToState = async () => { 
+		try {
+  			const value = await AsyncStorage.getItem('@DemoRNStore:settings_background');
+  			if (value !== null){
+    			this.setState({blackBackground: value == 'true'});
```
Comment (arakelyanva) [inline app/SettingsScreen.js:18]: value == 'true', always try to use 3 equation signs, e.g. value === 'true', compiles faster (no need for silent conversions) and gives less unpredictable errors

File: app/SettingsScreen.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  			await AsyncStorage.setItem('@DemoRNStore:settings_background', value.toString());
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('put background error');
+		}
+	}
+
+	onUserNameChanged = (text) => {
+		this.setState({username: text});
+		this.storeUserName(text);
```
Comment (arakelyanva) [inline app/SettingsScreen.js:51]: setState function has it's callback, so if you need something to happen exactly after state change, better option is to write: this.setState({ username: text }, () => this.storeUserName(text));

File: app/SettingsScreen.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		if (this.state.blackBackground) {
+			container.backgroundColor = 'black';
+		}
+		return container;
+	}
+
+	render(){
+		let { blackBackground } = this.state;
+		return(
+		<View style={ this.containerStyle() }>
```
Comment (arakelyanva) [inline app/SettingsScreen.js:76]: changing component style by assigning function to it is not good. Better would be in this case to write:
style={this.state.blackBackground ? styles.blackBackground : styles.defaultBackground} and add those styles in StyleSheet.create().

File: app/SettingsScreen.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+				<Switch
+					style={styles.rightItem}
+					onValueChange = { this.toggleSwitch }
+					value = {blackBackground}
+				/>
+			</View>
+			<View style={styles.row}>
+				<Text style={styles.leftItem} >Input your name</Text>
+				<TextInput
+        			style={styles.rightItem, {borderColor: 'gray', borderWidth: 1, width: 150}}
```
Comment (arakelyanva) [inline app/SettingsScreen.js:88]: Why pass two separate style objects? The right one is static styles anyway, could have put it in styles.rightItem. I am not sure your styles.rightItem works here anyway.

**Review by arakelyanva — COMMENTED** [review]
Also try to delete the commented code before pushing, if you don't need it anymore.

File: app/SettingsScreen.js (hunk 1 above)
Comment (MusheghKh) [reply app/SettingsScreen.js:18]: It's say that is unnecessary? Or it's my english is bad?
https://stackoverflow.com/questions/263965/how-can-i-convert-a-string-to-boolean-in-javascript

**Pull request closed without merging**